### PR TITLE
Moving setitem to the base class

### PR DESCRIPTION
**Pull request by josenavas**
I just move the code to the base class and updated the SQL to make use of the class variables to fill the sql correctly.

### CODE REVIEW

File: qiita_db/metadata_template/test/test_prep_template.py [hunk 1]
```diff
@@ -220,9 +220,11 @@ def test_get_none(self):
...
     """Tests the PrepSample class"""
     def test_setitem(self):
-        """setitem raises an error (currently not allowed)"""
-        with self.assertRaises(QiitaDBNotImplementedError):
-            self.tester['barcodesequence'] = 'GTCCGCAAGTTA'
+        with self.assertRaises(QiitaDBColumnError):
+            self.tester['column that does not exist'] = 0.3
+        self.assertEqual(self.tester['center_name'], 'ANL')
+        self.tester['center_name'] = "FOO"
+        self.assertEqual(self.tester['center_name'], "FOO")
```
Comment (squirrelo) [inline qiita_db/metadata_template/test/test_prep_template.py]: This also needs a test for when a column has the wrong datatype, triggering the ValueError.

File: qiita_db/metadata_template/base_metadata_template.py [hunk 2]
```diff
@@ -308,17 +309,79 @@ def handler(x):
...
+        column : str
+            The column to update
+        value : str
+            The value to set. This is expected to be a str on the assumption
+            that psycopg2 will cast as necessary when updating.
+
+        Raises
+        ------
+        QiitaDBColumnError
+            If the column does not exist in the table
```
Comment (squirrelo) [inline qiita_db/metadata_template/base_metadata_template.py]: Can you add the ValueError raised when value and database column types are not compatible here?

**Comment by squirrelo** [comment]
Couple comments.

File: qiita_db/metadata_template/test/test_prep_template.py (hunk 1 above)
Comment (josenavas) [inline qiita_db/metadata_template/test/test_prep_template.py]: This is a great point! When I wrote the test I noticed that there is a "bug" on the code (it is just preventing a useful message, therefore quoted). I'm fixing it right now.

File: qiita_db/metadata_template/base_metadata_template.py (hunk 2 above)
Comment (josenavas) [inline qiita_db/metadata_template/base_metadata_template.py]: Done

File: qiita_db/metadata_template/base_metadata_template.py [hunk 3]
```diff
@@ -308,17 +309,79 @@ def handler(x):
...
+                conn_handler.execute(sql, (value, self._id))
+            except Exception as e:
+                column_type = conn_handler.execute_fetchone(
+                    """SELECT data_type
+                       FROM information_schema.columns
+                       WHERE column_name=%s AND table_schema='qiita'
+                    """, (column,))[0]
+                value_type = type(value).__name__
+
+                if column_type != value_type:
```
Comment (squirrelo) [inline qiita_db/metadata_template/base_metadata_template.py]: This seems to be comparing SQL types (column_type) and python type (value_type). Is that kosher?

**Comment by josenavas** [comment]
:boom: ready for another round of review!

**Comment by ElDeveloper** [comment]
:+1:

**Pull request merged**